lib/housekeeping.js

```diff
@@ -38,17 +38,109 @@ class HouseKeeper {
     this.tagger = tagger;
   }
 
-  async _handleVolumeData(volume) {}
+  _calculateVolumeTotals(volumes) {}
+
+  _submitVolumeTotals(totalsByCategory) {
+    // Assumption:
+    // totalsByCategory is organized as follows:
+    // { 
+    //   ${region}: [ 
+    //     {
+    //       ${vol-type}: {
+    //         active: {
+    //           gb: value,
+    //           count: value
+    //         },
+    //         unused: {
+    //           gb: value,
+    //           count: value
+    //         }
+    //       }
+    //     },
+    //     ...
+    //   ]
+    // }
+
+    //assert(typeof totalsByCategory === 'object');
+
+    //assert(Object.keys(totalsByCategory).length === 1);
+    let region =  Object.keys(totalsByCategory)[0];
+    
+    let volumeTypes = totalsByCategory[region];
+    //assert(typeof volumeTypes === 'object');
+    //assert(typeof volumeTypes.hasOwnProperty('length'));
+
+    volumeTypes.forEach(function(data) {
+      //assert(typeof data === 'object');
+      
+      //assert(Object.keys(data).length === 1);
+      let type = Object.keys(data)[0];
+
+      //assert(data.hasOwnProperty('active'));
+      //assert(data.hasOwnProperty('unused'));
+      let active = data.active;
+      let unused = data.unused;
+
+      //assert(typeof active === 'object');
+      //assert(active.hasOwnProperty('gb'));
+      //assert(active.hasOwnProperty('count'));
+
+      this.monitor.count('ebs-volumes:${region}:gb-usage', active.gb);
```

Comment:
Forgive me if this is me missing the point, but do these need to be separated on `volumeType` as well? It seems like what you have on line 101 should be up here maybe?

Reply:
This is something to ask @jhford about. In the UCOSP project document, John listed just this set of series as a starting point. It might make sense to add a couple more to distinguish between active and unused volumes of each type, as well.